libs/js/banglejs/E_showMenu_Q3.js

```diff
@@ -133,19 +133,19 @@
         l = g.setFont("6x15").wrapString(title,r.w-pad);
       g.setFontAlign(-1,0).drawString(l.join("\n"), r.x+12, r.y+H/2);
     },
-    select : function(idx) {
+    select : function(idx, touch) {
       if (idx<0) return back&&back(); // title
       var item = menu[keys[idx]];
       Bangle.buzz(20);
-      if ("function" == typeof item) item(l);
+      if ("function" == typeof item) item(l, touch);
       else if ("object" == typeof item) {
-        // if a bool, just toggle it
         if ("number" == typeof item.value) {
           showSubMenu(item, keys[idx]);
         } else {
+          // if a bool, just toggle it
           if ("boolean"==typeof item.value)
             item.value=!item.value;
-          if (item.onchange) item.onchange(item.value);
+          if (item.onchange) item.onchange(item.value, l, touch);
```

Comment:
I feel like including `l` probably isn't super important since it's returned by `showMenu` anyway, but if you think it should be included that's fine.

I'd just say in that case maybe you should include it in the other 2 calls to `item.onchange`? Sending `touch` for those might not be appropriate I guess as it won't be a touch on the main menu, but a different menu.

Reply:
> I feel like including `l` probably isn't super important since it's returned by `showMenu` anyway, but if you think it should be included that's fine.

Makes me wonder why `l` is passed on for `if ("function"==typeof item)` in the first place then?

> I'd just say in that case maybe you should include it in the other 2 calls to `item.onchange`? Sending `touch` for those might not be appropriate I guess as it won't be a touch on the main menu, but a different menu.

Hm.. Maybe that makes sense (and maybe `touch` as well). I get unsure of myself every time I think about this again and from different perspectives.

I've taken a step back for now. Might think/code on this later in the week.

Reply:
> why l is passed 

Yeah, I think the idea was the calling function would get the scroll level out of the menu easily (but it could have done that anyway). One bad side-effect is if you do `"menitem":showNewMenu` you get a memory leak as the old menu is passed as an argument, but I think realistically we can't easily remove it now unless it breaks something.

... so thinking about the memory leak I guess my gut feeling is to not pass `l`.

Reply:
I think there's no problem with removing `l` from being sent into the callback function, as I find in https://github.com/espruino/Espruino/issues/2567 .